### PR TITLE
cvm-image-rewriter: refine the cvm-image-rewriter

### PR DESCRIPTION
**Pull request by Yanbo0101**
- Refine the cvm-image-rewriter
- Add readme for each plugin
- Update the serial number of plugin

### CODE REVIEW

File: tools/cvm-image-rewriter/README.md [hunk 1]
```diff
@@ -46,11 +46,11 @@ There are following customization plugins in Pre-Stage providing customization t
 | 03-netplan | Customize the netplan.yaml | N |
 | 04-user-authkey | Add auth key for user login instead of password | N |
 | 05-readonly-data | Fix some file permission to ready-only | N |
+| 06-initrd-update | Update the initrd image | N |
 | 07-install-mvp-guest | Install MVP TDX guest kernel | Y |
 | 08-device-permission | Fix the permission for device node | Y |
 | 09-ccnp-uds-directory-permission | Fix the permission for CCNP UDS directory | Y |
-| 60-initrd-update | Update the initrd image | N |
-| 98-ima-enable-simple | Enable IMA (Integrity Measurement Architecture) feature | N |
+| 97-ima-enable-simple | Enable IMA (Integrity Measurement Architecture) feature | N |
```
Comment (ruomengh) [inline tools/cvm-image-rewriter/README.md:53]: How about change 97 to 10

File: tools/cvm-image-rewriter/README.md [hunk 2]
```diff
@@ -136,15 +136,15 @@ Before running the tool, please choose the plugins that are needed.You can skip
...
 
     ```
-    touch pre-stage/01-resize-image/NOT_RUN
+    touch plugins/01-resize-image/NOT_RUN
     ```
 
 If the guest image is used for CCNP deployment, it's recommended to run below plugin combination according to different initial guest image type.
 Others are not required by CCNP and can be skipped.
-|  Base image | 01  | 02  | 03  | 04  | 05 | 07 | 08 | 09 | 60 | 98 |
+|  Base image | 01  | 02  | 03  | 04  | 05 | 06 | 07 | 08 | 09 | 97 |
```
Comment (ruomengh) [inline tools/cvm-image-rewriter/README.md:144]: 97 -> 10

File: tools/cvm-image-rewriter/plugins/02-motd-welcome/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Motd welcome
+
+This plugin is used to set the information in `/etc/motd`, which will be displayed after successful login.
```
Comment (ruomengh) [inline tools/cvm-image-rewriter/plugins/02-motd-welcome/README.md:3]: If the user wants to customize the welcome information, they will need to update the content of "files/etc/motd".

File: tools/cvm-image-rewriter/plugins/03-netplan/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Netplan
+
+This plugin is used to add network configuration file netplan.yaml in the `/etc/netplan/` directory, which will configure the network for the guest image.
```
Comment (ruomengh) [inline tools/cvm-image-rewriter/plugins/03-netplan/README.md:3]: If the user wants to customize netplan.yaml, it needs to update the content of "/etc/netplan/netplan.yaml"

File: tools/cvm-image-rewriter/plugins/04-user-authkey/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# User authkey
+
+This plugin is used to set the SSH login user and public key using environment variables ${CVM_USER} and ${CVM_AUTH_KEY}.
+
+The ${CVM_USER} default value is cvm. And if ${CVM_AUTH_KEY} is not specified like below, then this plugin will be skipped.
```
Comment (ruomengh) [inline tools/cvm-image-rewriter/plugins/04-user-authkey/README.md:5]: What's the default username/password if this is not performed?

File: tools/cvm-image-rewriter/plugins/98-test/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Test
+
+This plugin is used to test cloud-init, files, pre-stage and post-stage.
```
Comment (ruomengh) [inline tools/cvm-image-rewriter/plugins/98-test/README.md:3]: It's better to clarify this is a testing plugin. Besides, it;s better to add the expected result running the plugin.

Comment (Yanbo0101) [reply tools/cvm-image-rewriter/plugins/98-test/README.md]: @ruomengh  Updated. Please review it again. Thanks.

**Review by ruomengh — APPROVED** [review]
LGTM

File: tools/cvm-image-rewriter/README.md [hunk 7]
```diff
@@ -136,15 +136,15 @@ Before running the tool, please choose the plugins that are needed.You can skip
...
 
     ```
-    touch pre-stage/01-resize-image/NOT_RUN
+    touch plugins/01-resize-image/NOT_RUN
     ```
 
 If the guest image is used for CCNP deployment, it's recommended to run below plugin combination according to different initial guest image type.
 Others are not required by CCNP and can be skipped.
-|  Base image | 01  | 02  | 03  | 04  | 05 | 07 | 08 | 09 | 60 | 98 |
+|  Base image | 01  | 02  | 03  | 04  | 05 | 06 | 07 | 08 | 09 | 10 |
```
Comment (HaokunX-intel) [inline tools/cvm-image-rewriter/README.md:144]: The prefix of the plugin should be same with its real name.

**Review by HaokunX-intel — CHANGES_REQUESTED** [review]

File: tools/cvm-image-rewriter/README.md (hunk 7 above)
Comment (Yanbo0101) [reply tools/cvm-image-rewriter/README.md]: Updated. Please review it again. Thanks.

File: tools/cvm-image-rewriter/README.md [hunk 8]
```diff
@@ -37,7 +37,7 @@ framework, and the whole flow was divided into three stages:
 
 ### 2.1 Existing Plugins
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:38]: There is a 97-test directory.  Would it be possible to have a "sample" plugin setup directory and section 2.2 point to that?

File: tools/cvm-image-rewriter/README.md [hunk 9]
```diff
@@ -46,28 +46,28 @@ There are following customization plugins in Pre-Stage providing customization t
...
-2. Pre-stage execution on the host: the `<plugin directory>/host_run.sh` will be
+2. Pre-stage execution on the host: the `<plugin directory>/pre-stage/host_run.sh` will be
 executed before cloud-init stage
 3. cloud-init customization: please put the config yaml into `<plugin directory>/cloud-init/cloud-config`,
 and put the scripts to `<plugin directory>/cloud-init/x-shellscript`
 
-Please refer [the sample plugin](/tools/cvm-image-rewriter/pre-stage/99-test/).
+Please refer [the sample plugin](/tools/cvm-image-rewriter/plugins/97-test/).
 
 ## 3. How to Run the tool
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:72]: I would change sentence 2.

"This tool can run on bare metal or within a virtual machine using nesting as detailed in section 3.4"

Comment (eadamsintel) [reply tools/cvm-image-rewriter/README.md:72]: Is it worth mentioning this runs on debian 10?  That seems like an OS no one uses very much.  I could be wrong.

File: tools/cvm-image-rewriter/README.md [hunk 10]
```diff
@@ -136,19 +136,22 @@ Before running the tool, please choose the plugins that are needed.You can skip
...
+|  Base image | 01  | 02  | 03  | 04  | 05 | 06 | 07 | 08 | 60 | 98 |
 |---|---|---|---|---|---|---|---|---|---|---|
-|  Ubuntu base image | | | | | Y| | Y| Y| | |
+|  Ubuntu base image | | | | | | Y| Y| Y| | |
 | TD enlightened image | | | | | | | Y| Y| | |
 
 **NOTE:**
-  - TD enlightened image means the image already has TDX kernel. If not, plugin 05 is needed to install TDX kernel.
-  - Plugin 08 and 09 prepares device permission for CCNP deployment.
+  - All plugins need to be executed in numerical order.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:150]: Is it the need to be or is it that they will be?

File: tools/cvm-image-rewriter/README.md [hunk 11]
```diff
@@ -136,19 +136,22 @@ Before running the tool, please choose the plugins that are needed.You can skip
...
+  - TD enlightened image means the image already has TDX kernel. If not, plugin 06 is needed to install TDX kernel.
+  - Plugin 7 and Plugin 8 need to be executed before deploying CCNP to provide device permissions for CCNP.
+  - Plugin 60 requires copying or generating all files to the root directory first. When users customize plugins, please ensure that the plugin number with this requirement is placed before 60.
+  - Plugin 98 needs to be executed after all other plugins have completed.The number of the user-customized plugin must be before 98.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:154]: Missing space after the period

Comment (eadamsintel) [reply tools/cvm-image-rewriter/README.md:154]: Also, in the section below about the tool parameters I would change the -n to the following

-n  Silent running for virt-install with no output

File: tools/cvm-image-rewriter/plugins/03-netplan/README.md [hunk 12]
```diff
@@ -0,0 +1,3 @@
+# Netplan
+
+This plugin is used to add network configuration file netplan.yaml in the `/etc/netplan/` directory, which will configure the network for the guest image. Users can customize netplan.yaml by updating the contents of `files/etc/netplan/netplan.yaml`
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/03-netplan/README.md:3]: I would change this to 

This plugin is used to add network a network configuration file `netplan.yaml` in the ...   I would also put code quotes around the other netplan.yaml

File: tools/cvm-image-rewriter/plugins/04-user-authkey/README.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
+# User authkey
+
+This plugin is used to set the SSH login user and public key using environment variables ${CVM_USER} and ${CVM_AUTH_KEY}.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/04-user-authkey/README.md:3]: I think it is worth mentioning this won't run unless those two environmental variables are set.

File: tools/cvm-image-rewriter/plugins/05-readonly-data/pre-stage/host_run.sh [hunk 14]
```diff
@@ -3,7 +3,7 @@
 DIR=$(dirname "$(readlink -f "$0")")
 FILE_LIST="$DIR/file_list"
 CLD_SH_READONLY_FILE="01-file-readonly.sh"
-CLD_SH="$DIR/cloud-init/x-shellscript/$CLD_SH_READONLY_FILE"
+CLD_SH="$DIR/../cloud-init/x-shellscript/$CLD_SH_READONLY_FILE"
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/05-readonly-data/pre-stage/host_run.sh:6]: I tried to comment on the readme.md but it wont let me because only the name was changed.  My comment is how to users know how to enable this or not?  Is it that just by adding files to this then this plugin runs?

File: tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md [hunk 15]
```diff
@@ -0,0 +1,26 @@
+# Install TDX guest kernel
+
+This plugin is used to install TDX guest kernel from a given local repository.
+
+- Prepare the local repository.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md:5]: This text seems redundant

File: tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+
+Then you can place this local repository in the `pre-stage/artifacts/` directory.(Optional) 
+```
+mkdir -p ./pre-stage/artifacts
+mv <your guest repo> ./pre-stage/artifacts/
+```
+
+- Set the environment variable.
+
+Set ${CVM_TDX_GUEST_REPO} to the repository absolute path, or this plugin will be skipped.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md:17]: put `CVM_TDX_GUEST_REPO` in code block quotes

File: tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md [hunk 17]
```diff
@@ -0,0 +1,26 @@
+# Install TDX guest kernel
+
+This plugin is used to install TDX guest kernel from a given local repository.
+
+- Prepare the local repository.
+
+Prepare the local repository and confirm that there are Debian packages related to the TDX kernel in the `/jammy/amd64/` directory of this repository.
+
+Then you can place this local repository in the `pre-stage/artifacts/` directory.(Optional)
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md:9]: I would remove Optional and have a clearer description up top of when this needs to run and not run and how to not run it.

Comment (eadamsintel) [reply tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md:9]: I saw further down you did explain how to not run.  Perhaps what is needed is a standard format for each plugin for how the readme is laid out. 

section 1 - What is the plugin
section 2 - How to enable or disable and whether it is optional
section 3 - How to configure the plugin and install

File: tools/cvm-image-rewriter/plugins/07-device-permission/README.md [hunk 18]
```diff
@@ -0,0 +1,3 @@
+# Device permission
+
+This plugin is used to setup udev rules for CCNP device plugin. Users do not need to customize.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/07-device-permission/README.md:3]: Why make it a plugin then?  Why not just set this as part of the main install script?  Unless you think people want to customize udev rules in the future I would remove this option.  You can paralyze users by choice if you give too many options.

File: tools/cvm-image-rewriter/plugins/08-ccnp-uds-directory-permission/README.md [hunk 19]
```diff
@@ -0,0 +1,3 @@
+# CCNP UDS directory permission
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/08-ccnp-uds-directory-permission/README.md:1]: Same comment as above. If users don't need to customize why give the option and confuse them?

File: tools/cvm-image-rewriter/plugins/97-test/README.md [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+This is a testing plugin used to test cloud-init, files, pre-stage and post-stage.
+
+Example Output:
+```
+# output of the cloud-init
+=========== example for customized x-shellscript ===========
+
+#  output of the pre-stage
+pre-stage host_run haha
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/97-test/README.md:11]: haha at the "haha"

**Review by eadamsintel — COMMENTED** [review]
I don't have time right now to actually go through the scripts and try them out so I focused on the readme.md for each repo to suggest ideas for structure and clarity.

File: tools/cvm-image-rewriter/README.md (hunk 10 above)
Comment (Yanbo0101) [reply tools/cvm-image-rewriter/README.md:151]: Thanks for your review. It is need to be and it will be, for example, plugin 98 cannot be the first one to execute.

File: tools/cvm-image-rewriter/plugins/04-user-authkey/README.md (hunk 13 above)
Comment (Yanbo0101) [reply tools/cvm-image-rewriter/plugins/04-user-authkey/README.md]: Because `CVM_USER` has default values, while `CVM_AUTH_KEY` does not, I have clarified in the next line that if the value for `CVM_AUTH_KEY` is not set, this plugin will be skipped.

File: tools/cvm-image-rewriter/plugins/05-readonly-data/pre-stage/host_run.sh (hunk 14 above)
Comment (Yanbo0101) [reply tools/cvm-image-rewriter/plugins/05-readonly-data/pre-stage/host_run.sh:6]: Yes, unless you have added a `NOT_RUN` file in this plugin directory following the `tools/cvm-image-rewriter/README.md ` instructions.

File: tools/cvm-image-rewriter/plugins/07-device-permission/README.md (hunk 18 above)
Comment (Yanbo0101) [reply tools/cvm-image-rewriter/plugins/07-device-permission/README.md:3]: In fact, the plugin form is more convenient for flexible operations and maintenance. Users can customize udev rules in the future as TDX is updated or device-plugin is updated, but currently, we do not recommend doing so.

**Comment by Yanbo0101** [comment]
@eadamsintel Thanks for your review! Most of the comments updated Could you please help review it again?

**Review by HaokunX-intel — APPROVED** [review]
LGTM

File: tools/cvm-image-rewriter/README.md [hunk 21]
```diff
@@ -46,28 +46,29 @@ There are following customization plugins in Pre-Stage providing customization t
...
 1. File override: all files under `<plugin directory>/files` will be copied the
 corresponding directory in target guest image.
-2. Pre-stage execution on the host: the `<plugin directory>/host_run.sh` will be
+2. Pre-stage execution on the host: the `<plugin directory>/pre-stage/host_run.sh` will be
 executed before cloud-init stage
 3. cloud-init customization: please put the config yaml into `<plugin directory>/cloud-init/cloud-config`,
 and put the scripts to `<plugin directory>/cloud-init/x-shellscript`
 
-Please refer [the sample plugin](/tools/cvm-image-rewriter/pre-stage/99-test/).
+Please refer [the sample plugin](/tools/cvm-image-rewriter/plugins/97-sample/).
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:71]: Please refer to

File: tools/cvm-image-rewriter/README.md [hunk 22]
```diff
@@ -76,7 +77,7 @@ Please refer [the sample plugin](/tools/cvm-image-rewriter/pre-stage/99-test/).
 1. This tool has been tested on `Ubuntu 22.04` and `Debian 10`. It is recommend to use
 `Ubuntu 22.04`.
 
-2. This tool can run on bare metal or virtual machine (with nest VM like `Intel VT-x`, detailed in [Section 3.4](#3.4-Run-in-Nested-VM-(Optional)))
+2. This tool can run on bare metal or within a virtual machine using nesting as detailed in [Section 3.4](#3.4-Run-in-Nested-VM-(Optional))
 
 3. Please install following packages on Ubuntu/Debian:
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:82]: I would reword this as  "Please install the following"

Comment (eadamsintel) [reply tools/cvm-image-rewriter/README.md:82]: For 1 through 7 some of these end with a period and some do not. I would make it consistent.

Comment (eadamsintel) [reply tools/cvm-image-rewriter/README.md:82]: Further below in number 7 I would change it to   "If it is needed to customize the connection URL, you can"

Comment (eadamsintel) [reply tools/cvm-image-rewriter/README.md:82]: Also, I would change this and make it its own sentence "Please make sure the current user belongs to the libvirt group via the following commands"

File: tools/cvm-image-rewriter/README.md [hunk 23]
```diff
@@ -136,19 +137,22 @@ Before running the tool, please choose the plugins that are needed.You can skip
 For example:
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:137]: Right above I would reword to  "under the current directory"

File: tools/cvm-image-rewriter/README.md [hunk 24]
```diff
@@ -136,19 +137,22 @@ Before running the tool, please choose the plugins that are needed.You can skip
...
 
     ```
-    touch pre-stage/01-resize-image/NOT_RUN
+    touch plugins/01-resize-image/NOT_RUN
     ```
 
 If the guest image is used for CCNP deployment, it's recommended to run below plugin combination according to different initial guest image type.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:143]: I would reword to "it's recommended to run the below plugin combinations depending on which guest image type is used."

File: tools/cvm-image-rewriter/README.md [hunk 25]
```diff
@@ -136,19 +137,22 @@ Before running the tool, please choose the plugins that are needed.You can skip
...
 Others are not required by CCNP and can be skipped.
-|  Base image | 01  | 02  | 03  | 04  | 05 | 07 | 08 | 09 | 60 | 98 |
+|  Base image | 01  | 02  | 03  | 04  | 05 | 06 | 07 | 08 | 60 | 98 |
 |---|---|---|---|---|---|---|---|---|---|---|
-|  Ubuntu base image | | | | | Y| | Y| Y| | |
+|  Ubuntu base image | | | | | | Y| Y| Y| | |
 | TD enlightened image | | | | | | | Y| Y| | |
 
 **NOTE:**
-  - TD enlightened image means the image already has TDX kernel. If not, plugin 05 is needed to install TDX kernel.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:150]: "already has a TDX kernel. If not, plugin 05 is required to install a TDX kernel"

File: tools/cvm-image-rewriter/README.md [hunk 26]
```diff
@@ -163,7 +167,7 @@ Optional
   -s <connection socket>    Default is connection URI is qemu:///system,
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/README.md:167]: Another thing I noticed is in the run.sh output I would slightly change the wording of the "-s" I would say "Default connection URS is qemu:///system" Is it supposed to be three ///?

Comment (eadamsintel) [reply tools/cvm-image-rewriter/README.md:167]: Further below you saw the word firstly.  That isn't that common of a word.  I would just use first.  That isn't super important though.

File: tools/cvm-image-rewriter/plugins/03-netplan/README.md [hunk 27]
```diff
@@ -0,0 +1,3 @@
+# Netplan
+
+This plugin is used to add network configuration file `netplan.yaml` in the `/etc/netplan/` directory, which will configure the network for the guest image. Users can customize netplan.yaml by updating the contents of `files/etc/netplan/netplan.yaml`
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/03-netplan/README.md:3]: Minor nit   add a code block formatting to the second `netplan.yaml`

File: tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md [hunk 28]
```diff
@@ -0,0 +1,22 @@
+# Install TDX guest kernel
+
+This plugin is used to install TDX guest kernel from a given local repository.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md:3]: "install a TDX guest"

File: tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md [hunk 29]
```diff
@@ -0,0 +1,22 @@
...
+
+# Prerequisite
+
+Prepare the local repository and confirm that there are Debian packages related to the TDX kernel in the `/jammy/amd64/` directory of this repository. You can choose to place this local repository in the `pre-stage/artifacts/` directory.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md:7]: Maybe instead of  "you can choose to place" to make it  "It is recommended to place"

Comment (eadamsintel) [reply tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md:7]: Or is it required to be here?  If it is required to be in this spot for this to work I would use stronger language telling users it is required.

File: tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md [hunk 30]
```diff
@@ -0,0 +1,22 @@
...
+Set `${CVM_TDX_GUEST_REPO}` to the repository absolute path, or this plugin will be skipped. 
+```
+export CVM_TDX_GUEST_REPO=$(pwd)/pre-stage/artifacts/<your guest repo>
+
+# Or
+export CVM_TDX_GUEST_REPO=<your local guest repo>
+```
+
+
+_NOTE: IF the original image is smaller than 1.5G, please also set the environment variable GUEST\_SIZE larger ._
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md:22]: It might be worth mentioning this will cause plugin 01 to run by setting this variable.

File: tools/cvm-image-rewriter/plugins/07-device-permission/README.md (hunk 18 above)
Comment (eadamsintel) [reply tools/cvm-image-rewriter/plugins/07-device-permission/README.md:3]: I think then I am confused by the  "Users do not need to customize"   Maybe it would be better to say   "It is typically not required for users to customize unless additional udev rules are needed."

File: tools/cvm-image-rewriter/plugins/08-ccnp-uds-directory-permission/README.md [hunk 31]
```diff
@@ -0,0 +1,3 @@
...
+
+This plugin is used to prepare shared Unix domain socket directories for CCNP. Users do not need to customize.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/08-ccnp-uds-directory-permission/README.md:3]: I would recommend the same language as above to indicate they don't need to do this but then give the reason they might.

File: tools/cvm-image-rewriter/plugins/60-initrd-update/README.md [hunk 32]
```diff
@@ -0,0 +1,3 @@
+# Initrd update
+
+This plugins is used to update the initrd image. Users place executable files in the `files/` directory, and the plugin will copy these executable files to the initramfs.
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/plugins/60-initrd-update/README.md:3]: Is an initrd considered an executable file?   Would it be better to say  something like  "To change the initrd image, place the necessary files in the `files/` directory. The plugin will copy these files to the initramfs."

File: tools/cvm-image-rewriter/run.sh [hunk 33]
```diff
@@ -41,15 +36,36 @@ Optional
...
   -n                        Silence running for virt-install, no output
-  -h                        Show usage
 
 EOM
 }
 
+#
+# Update plugin directory list
+#
+# If want to skip some steps, please create a file named "NOT_RUN"
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/run.sh:46]: I would reword  "To skip the installation of a specific plugin place a file called "NOT_RUN" in each plugin directory to be skipped"

File: tools/cvm-image-rewriter/run.sh [hunk 34]
```diff
@@ -41,15 +36,36 @@ Optional
...
+        fi
+    done
+    plugin_dirs=("${tmp_dirs[@]}")
+}
+
 #
 # Prepare the files copying to target guest image.
 #
-# 1. Scan following content from all subdirectories under pre-stage
+# 1. Scan following content from all subdirectories under plugins
```
Comment (eadamsintel) [inline tools/cvm-image-rewriter/run.sh:68]: "Scan the following content"

**Review by eadamsintel — COMMENTED** [review]
I reread the entire README.md from the beginning and added some grammar comments. I also added some suggested rewrites for better clarify for the user.

File: tools/cvm-image-rewriter/README.md (hunk 26 above)
Comment (Yanbo0101) [reply tools/cvm-image-rewriter/README.md]: > Another thing I noticed is in the run.sh output I would slightly change the wording of the "-s" I would say "Default connection URS is qemu:///system" Is it supposed to be three ///?

Yes, you can refer to https://libvirt.org/drvqemu.html#connections-to-qemu-driver, and according to this link it seems better to use `URI`.

File: tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md (hunk 29 above)
Comment (Yanbo0101) [reply tools/cvm-image-rewriter/plugins/06-install-tdx-guest-kernel/README.md]: It's not required，recommended is fine. Thanks.

**Comment by Yanbo0101** [comment]
@eadamsintel All files have been updated. Thank you very much for your review comments.

**Comment by Yanbo0101** [comment]
@eadamsintel Thanks for your review. Since this repo is being moved to a new location, I will merge this PR first. Thanks again.
